lib/chef/knife/ec2_server_create.rb

```diff
@@ -689,6 +689,8 @@ def bootstrap_common_params(bootstrap)
         bootstrap.config[:first_boot_attributes_from_file] = locate_config_value(:first_boot_attributes_from_file)
         bootstrap.config[:encrypted_data_bag_secret] = s3_secret || locate_config_value(:secret)
         bootstrap.config[:encrypted_data_bag_secret_file] = locate_config_value(:secret_file)
+        #cl_secret needs to be set as chef checks for this key
+        Chef::Config[:knife][:cl_secret] = s3_secret if locate_config_value(:s3_secret)
```

Comment:
@dheerajd-msys please improve the comment here. can you answer "why does chef need this key?" in the comment?